internal/lifecycle/readme.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lifecycle
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/util"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// The tag that should appear immediately before code blocks in a README to indicate that the enclosed commands
+// are to be used by this program for building and deploying the sample.
+var codeTag = "sst-run-linuxmacos"
+
+// parseREADME parses a README file with the given name. It reads terminal commands surrounded by one of the codeTags
+// listed above and loads them into a Lifecycle. In the process, it replaces the Cloud Run service name and Container
+// Registry tag with the provided inputs.
+func parseREADME(filename, serviceName, gcrURL string) (Lifecycle, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	var lifecycle Lifecycle
+
+	inCode := false
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		if strings.Contains(line, "```") && inCode {
+			inCode = false
+		} else if strings.Contains(line, codeTag) {
```

Comment:
As mentioned above (https://github.com/GoogleCloudPlatform/serverless-sample-tester/pull/2#discussion_r465461157), it appears there are multiple syntaxes for line comments in Mardown. That's why the liberal option of just looking for a mention the specific code tag in the line is used. I'm not sure whether we should just require this line specifically.

And if we stick with the initial implementation, after some thought, I'm now thinking that we shouldn't thrown an error if we aren't able to find the start of a code block on the next line; we should just ignore the current line with a code tag.

Reply:
Also, this, I believe, is the same way that the releasetool looks for package version update tags/markers in READMEs like the ones found here:

https://raw.githubusercontent.com/googleapis/java-bigtable/master/README.md

releasetool source: https://github.com/googleapis/releasetool/blob/master/releasetool/commands/start/java.py#L229

Reply:
Looks like the source you've linked matches `{<some_tag>}` (not just `<some_tag>`). I think it would make sense to be more explicit (again to minimize false positives)

Reply:
Okay, I've update the `codeTag` constant to be `{sst-run-unix}`